msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IMDSManagedIdentity.java

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class IMDSManagedIdentity extends AbstractManagedIdentitySource{
+
+    // IMDS constants. Docs for IMDS are available here https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http
+    private final static Logger LOG = LoggerFactory.getLogger(IMDSManagedIdentity.class);
+    private static URI DEFAULT_IMDS_ENDPOINT;
+
+    static {
+        try {
+            DEFAULT_IMDS_ENDPOINT = new URI("http://169.254.169.254/metadata/identity/oauth2/token");
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String imdsTokenPath = "/metadata/identity/oauth2/token";
```

Comment:
1. Please review the "static final" modifiers throughout. Is there an analyzer that can help @Avery-Dunn ?
2. Consider having all the constants and fields in one place

Reply:
If you're talking about analyzers to find minor issues/make code suggestions I use IntelliJ's 'SonarLint' plugin, it's pretty good at making best-practices suggestions out of the box.

MSAL Java is already riddled with stuff like 'static final' vs. 'final static' so it's not too big of an issue, but I've been fixing those sorts of things whenever SonarLint highlights them in a file I was changing anyway.